### PR TITLE
continuing scrapping even if article doesn't render

### LINKED ISSUE #1419: Unable to scrape wiktionary CHR any more
https://farm.openzim.org/recipes/wiktionary_chr

Seems to be a 1.11.4 regression

### PR DESCRIPTION
**Pull request by MananJethwani**
this resolves #1419
@kelson42 these articles give 200 response so we needed to check if they have any content or not if they don't have any I am returning a new `RENDERING_ERROR` .

### CODE REVIEW

**Comment by kelson42** [comment]
@MananJethwani would you be able to share here the url which generates the rendering error please?

**Comment by MananJethwani** [comment]
@kelson42 here it is -> https://chr.wiktionary.org/w/api.php?action=visualeditor&mobileformat=html&format=json&paction=parse&page=likaledes
the content is not there so it couldn't be rendered

**Comment by MananJethwani** [comment]
@kelson42 please review this I have rebased it.

**Review by kelson42 — CHANGES_REQUESTED** [review]
The problem with this PR is:
* I don't have a clear/simple reproducation case
* The given explanation (in the ticket) seems wrong because it always dies, so this is not a sporadic problem. I would like to understand the problem
* I don't understand the solution... probably because I don't understand what is the problem
* There is no automated test

**Review by kelson42 — CHANGES_REQUESTED** [review]
This case seems to happen when an article is deleted in the meantime between ns article list retrieval and content retrieval. This is a valid scenario. We could check the `oldid` to verify this case and skip (with a log message in such a case). Automated tests have to be adatped.

**Comment by MananJethwani** [comment]
@kelson42 are these checks fine?

File: test/unit/saveArticles.test.ts [hunk 1]
```diff
@@ -261,3 +260,16 @@ test('treat multiple subtitles in one video', async(t) => {
         'Video multiple subtitles rewriting matches');
     t.equals(contentRes.subtitles.length, 4, 'All subtitles are found for this video');
 });
+
+test('Desktop article Rendering Test' , async(t) => {
```
Comment (kelson42) [inline test/unit/saveArticles.test.ts:264]: Name of the test is wrong

File: test/unit/saveArticles.test.ts [hunk 2]
```diff
@@ -261,3 +260,16 @@ test('treat multiple subtitles in one video', async(t) => {
...
+    const DELETED_ARTICLE_FAIL_MESSAGE = 'Throwing error if article is deleted';
```
Comment (kelson42) [inline test/unit/saveArticles.test.ts:265]: You should no redeclare the error message (which btw is not the same as in the code). Yo shoukd test that the exception is gneraged properly.

**Review by kelson42 — CHANGES_REQUESTED** [review]

File: test/unit/saveArticles.test.ts (hunk 2 above)
Comment (MananJethwani) [reply test/unit/saveArticles.test.ts:265]: @kelson42 this is not the message used for error, it is the message shown in the terminal when the integration test fails or passes, I had to put it in 2 places one if `renderDesktopArticle()` didn't throw any error it should fail, and once in the catch block if the error is different.

Comment (kelson42) [reply test/unit/saveArticles.test.ts:265]: Ok, but you should just success if proper exception... without having to create a fake exception.

**Review by kelson42 — CHANGES_REQUESTED** [review]
Implement the test is a better way to avoid creating a fake exception.

**Comment by MananJethwani** [comment]
@kelson42 is this fine now?

**Review by kelson42 — APPROVED** [review]

**Pull request merged**